### PR TITLE
Deck building

### PR DESCRIPTION
**Pull request by StephenGibbsDev**
- added dropdown on details page for deck selection
- simple + crude adding to deck functionality 
- a few changes to db context as to only have one in the repo that handles identity + db

### CODE REVIEW

File: src/MTGCore/Controllers/CardController.cs [hunk 1]
```diff
@@ -53,16 +58,33 @@ public async Task<ActionResult> Details(int id)
...
 
-            //string rootPath = _env.WebRootPath;
 
             response.manaCost = _conversion.ConvertToSymbol(response.manaCost);
 
-            var model = _mapper.Map<Cards>(response);
+            var model = _mapper.Map<CardDto>(response);
+
+            var UserIDString = _userManager.GetUserId(HttpContext.User);
+
```
Comment (ConnorDownes) [inline src/MTGCore/Controllers/CardController.cs]: I like the idea of a user manager class. Seems easy to extend in the future.

File: src/MTGCore/Controllers/CardController.cs [hunk 2]
```diff
@@ -53,16 +58,33 @@ public async Task<ActionResult> Details(int id)
...
+            var userId = new Guid(UserIDString);
+
```
Comment (ConnorDownes) [inline src/MTGCore/Controllers/CardController.cs]: Is storing IDs as GUIDs better than storing them as integers?

File: src/MTGCore/Controllers/DeckController.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+        [HttpPost]
+        public IActionResult Create(Deck deck)
+        {
+            var UserIDString = _userManager.GetUserId(HttpContext.User);
+
+            deck.UserID = new Guid(UserIDString);
+
+            _context.Deck.Add(deck);
+
```
Comment (ConnorDownes) [inline src/MTGCore/Controllers/DeckController.cs]: Is this a direct call to the context? In which case, would it eventually be refactored and put into a service or repository?

Comment (StephenGibbsDev) [reply src/MTGCore/Controllers/DeckController.cs]: Yes, have no idea how to implement this but will put it in an issue.

File: src/MTGCore/Controllers/CardController.cs (hunk 2 above)
Comment (StephenGibbsDev) [reply src/MTGCore/Controllers/CardController.cs]: no its just that microsofts identity is set up to use GUIDS for some reason.

https://dba.stackexchange.com/a/266

good post on pros and cons. May Refactor in the future if its easy to do with identity

**Pull request merged**